tree/treeplayer/inc/ROOT/TVec.hxx

```diff
@@ -484,7 +468,13 @@ template <class T>
 std::ostream &operator<<(std::ostream &os, const TVec<T> &v)
 {
    // In order to print properly, convert to 64 bit int if this is a char
-   using Print_t = typename std::conditional<ROOT::Internal::VecOps::TIsChar<T>::value, long long int, T>::type;
+   constexpr bool mustConvert = std::is_same<char, T>::value ||
+                                std::is_same<signed char, T>::value ||
+                                std::is_same<unsigned char, T>::value ||
+                                std::is_same<wchar_t, T>::value ||
+                                std::is_same<char16_t, T>::value ||
+                                std::is_same<char32_t, T>::value;
+   using Print_t = typename std::conditional<mustConvert, long long int ,T>::type;
```

Comment:
why do we need to convert `char` to `long long int` when printing the contents of `TVec<char>`?

Reply:
Because we assume these chars are numbers (VecOps are mainly intended for analysis and not text manipulation) and this is a comfortable way to print the value of these numbers.

Reply:
ok, in other words we assume that physicists/experiments predominantly use arrays of chars as memory-efficient arrays of short integers, and print them as such.